src/java/org/jivesoftware/openfire/spi/RoutingTableImpl.java

```diff
@@ -454,36 +454,49 @@ private boolean routeToComponent(JID jid, Packet packet,
 	 * @return <tt>true</tt> if the packet was routed successfully,
 	 *         <tt>false</tt> otherwise.
 	 */
-	private boolean routeToRemoteDomain(JID jid, Packet packet,
-			boolean routed) {
-		byte[] nodeID = serversCache.get(jid.getDomain());
-		if (nodeID != null) {
-		    if (server.getNodeID().equals(nodeID)) {
-		        // This is a route to a remote server connected from this node
-		        try {
-		            localRoutingTable.getRoute(jid.getDomain()).process(packet);
-		            routed = true;
-		        } catch (UnauthorizedException e) {
-		            Log.error("Unable to route packet " + packet.toXML(), e);
-		        }
-		    }
-		    else {
-		        // This is a route to a remote server connected from other node
-		        if (remotePacketRouter != null) {
-		            routed = remotePacketRouter.routePacket(nodeID, jid, packet);
-		        }
-		    }
-		}
-		else if (!RemoteServerManager.canAccess(jid.getDomain())) { // Check if the remote domain is in the blacklist
-            Log.info( "Will not route: Remote domain {} is not accessible according to our configuration (typical causes: server federation is disabled, or domain is blacklisted).", jid.getDomain() );
-            routed = false;
+	private boolean routeToRemoteDomain(JID jid, Packet packet, boolean routed)
+    {
+		if ( JiveGlobals.getBooleanProperty( ConnectionSettings.Server.ALLOW_ANONYMOUS_OUTBOUND_DATA, false ) )

```

Comment:
Also, the default here is false, I thought from your PR message that it was defaulted to be true?  Sorry, maybe I am not awake yet this fine morning!